code/game/objects/items/weapons/clown_items.dm

```diff
@@ -146,6 +146,42 @@
 	hitsound = 'sound/items/quack.ogg'
 	honk_delay = 10
 
+#define TELE_COOLDOWN 5 SECONDS
+
+/obj/item/weapon/bikehorn/rubberducky/quantum
+	desc = "A quantum quacker"
+	var/teleport_range = 5
+	var/last_teleport
+
+/obj/item/weapon/bikehorn/rubberducky/quantum/New()
+	..()
+	processing_objects.Add(src)
+
+/obj/item/weapon/bikehorn/rubberducky/quantum/Destroy()
+	processing_objects.Remove(src)
+	..()
+
+/obj/item/weapon/bikehorn/rubberducky/quantum/process()
+	var/visible = FALSE
+
+	for (var/mob/living/M in viewers(src))
+		if(!M.isUnconscious() && !is_blind(M))
+			visible = TRUE
+			break
+
+	if(!visible && world.time > last_teleport + TELE_COOLDOWN)
```

Comment:
It might sound like a meme but I was actually serious
Someday, someone will spawn 200 of these and this could be the difference between the server chugging and being mostly OK